src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -275,7 +280,7 @@ private void WriteClientCtors(CodeWriter writer, LowLevelRestClient client, Buil
             {
                 WriteConstructor(writer, client, CredentialKind.Token, context);
             }
-            if (context.Configuration.CredentialTypes.Length == 0)
+            if (!context.CodeModel.Security.Schemes.Any())
```

Comment:
Can we use the same foreach pattern as dataplane uses here?

Reply:
Maybe, but there is a difference - DataPlaneClientWriter doesn't have a fallback "if none of these, do this other one". Here we want to only generate the no-auth constructor if no other one was created. 

I could use a foreach and a bool, but that felt worse to me.

Reply:
Yeah, but LLC has 3 patterns now: foreach loops, bools, and enum. Seems like too many ways to talk about the same thing.

Reply:
Fair enough. Let me look